hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/ClientConfigXmlGenerator.java

```diff
@@ -0,0 +1,708 @@
+                   "max-size-policy", eviction.getMaximumSizePolicy(),
+                   "eviction-policy", eviction.getEvictionPolicy())
+           .node("preloader", null, "enabled", preloader.isEnabled(),
+                   "directory", preloader.getDirectory(),
+                   "store-initial-delay-seconds", preloader.getStoreInitialDelaySeconds(),
+                   "store-interval-seconds", preloader.getStoreIntervalSeconds());
+        //close near-cache
+        gen.close();
+    }
+
+    private static String classNameOrImplClass(String className, Object impl) {
+        return !isNullOrEmpty(className) ? className
+                : impl != null ? impl.getClass().getName()
+                : null;
+    }
+
+    private static final class XmlGenerator {
+
+        private final StringBuilder xml;
+        private final ArrayDeque<String> openNodes = new ArrayDeque<String>();
+
+        private XmlGenerator(StringBuilder xml) {
+            this.xml = xml;
+        }
+
+        XmlGenerator open(String name, Object... attributes) {
+            appendOpenNode(xml, name, attributes);
+            openNodes.addLast(name);
+            return this;
+        }
+
+        XmlGenerator node(String name, Object contents, Object... attributes) {
+            appendNode(xml, name, contents, attributes);
+            return this;
+        }
+
+        XmlGenerator close() {
+            appendCloseNode(xml, openNodes.pollLast());
+            return this;
+        }
+
+        XmlGenerator appendProperties(Properties props) {
+            if (!props.isEmpty()) {
+                open("properties");
+                Set keys = props.keySet();
+                for (Object key : keys) {
+                    node("property", props.getProperty(key.toString()), "name", key.toString());
+                }
+                close();
+            }
+            return this;
+        }
+
+        XmlGenerator appendProperties(Map<String, Comparable> props) {
+            if (!MapUtil.isNullOrEmpty(props)) {
+                open("properties");
+                for (Map.Entry<String, Comparable> entry : props.entrySet()) {
+                    node("property", entry.getValue(), "name", entry.getKey());
+                }
+                close();
+            }
+            return this;
+        }
+
+        private static void appendOpenNode(StringBuilder xml, String name, Object... attributes) {
+            xml.append('<').append(name);
+            appendAttributes(xml, attributes);
+            xml.append('>');
+        }
+
+        private static void appendCloseNode(StringBuilder xml, String name) {
+            xml.append("</").append(name).append('>');
+        }
+
+        private static void appendAttributes(StringBuilder xml, Object... attributes) {
+            for (int i = 0; i < attributes.length; ) {
+                xml.append(" ");
+                escapeXml(attributes[i++], xml);
```

Comment:
We don't need to escape attribute names. Rather, we should check them for sanity, I found [these rules](https://stackoverflow.com/a/44057262/952135). I think we can just ignore that, we don't use user-input for attribute names and hard-coded attribute names should be ok.